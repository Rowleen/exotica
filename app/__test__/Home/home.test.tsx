import '@testing-library/jest-dom'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render, screen } from '@testing-library/react'
import Home from '../../page'

const queryClient = new QueryClient()

describe('Home', () => {
  it('renders homepage unchanged', () => {
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <Home />
      </QueryClientProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders a heading', () => {
    render(
      <QueryClientProvider client={queryClient}>
        <Home />
      </QueryClientProvider>
    )

    const heading = screen.getByRole('heading', { level: 1 })

    expect(heading).toBeInTheDocument()
  })
})
