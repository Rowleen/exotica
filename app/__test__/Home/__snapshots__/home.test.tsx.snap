// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div
    class="home"
  >
    <header
      class="header"
    >
      <div
        class="logoWrapper"
      >
        <img
          alt="Logo image"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/images/logo.svg"
          style="color: transparent;"
          width="40"
        />
      </div>
      <button
        class="button secondary medium"
        type="button"
      >
        Create new trip
      </button>
    </header>
    <section
      class="titleWrapper"
    >
      <h1
        class="title"
      >
        The places you dream of
      </h1>
      <h2
        class="subtitle"
      >
        Let's live new adventures
      </h2>
    </section>
    <section
      class="searcherWrapper"
    >
      <form
        class="searchForm"
      >
        <input
          class="input"
          placeholder="Search trips"
        />
        <div
          class="buttonWrapper"
        >
          <button
            class="button primary small"
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </section>
    <section
      class="filterPillsWrapper"
    >
      <button
        class="pill"
        type="button"
        value="all"
      >
        All
      </button>
      <button
        class="pill"
        type="button"
        value="toDo"
      >
        Upcoming
      </button>
      <button
        class="pill"
        type="button"
        value="done"
      >
        Complete
      </button>
    </section>
    <main
      class="tripsWrapper"
    >
      Spinner
    </main>
  </div>
</div>
`;
